sound/soc/sof/pcm.c

```diff
@@ -913,7 +913,8 @@ void snd_sof_new_platform_drv(struct snd_sof_dev *sdev)
 	pd->compress_ops = &sof_probe_compressed_ops;
 #endif
 	pd->pcm_construct = sof_pcm_new;
-	pd->ignore_machine = drv_name;
+	if (!plat_data->desc->no_ignore_machine)
+		pd->ignore_machine = drv_name;
```

Comment:
I would go with @dbaluta's suggestion. Call the flag the same in your descriptor as in the ASoC driver definition, this will avoid logical inversions that are unfortunately oh so common in engineering circles.